scripts/zones/Batallia_Downs_[S]/mobs/Dark_Ixion.lua

```diff
@@ -1,11 +1,69 @@
 -----------------------------------
 -- Area: Batallia_Downs_[S]
---  Mob: Dark Ixion
+--   NM: Dark Ixion
+-----------------------------------
+require('scripts/globals/dark_ixion')
 -----------------------------------
 local entity = {}
 
 entity.onMobDeath = function(mob, player, optParams)
-    player:addTitle(xi.title.IXION_HORNBREAKER)
+    xi.darkixion.onMobDeath(mob, player, optParams)
+end
+
+entity.onMobDespawn = function(mob)
+    xi.darkixion.onMobDespawn(mob)
+end
+
+entity.onCriticalHit = function(mob)
+    xi.darkixion.onCriticalHit(mob)
+end
+
+entity.onWeaponskillHit = function(mob, attacker, weaponskill)
+    xi.darkixion.onWeaponskillHit(mob, attacker, weaponskill)
+end
+
+entity.onMobInitialize = function(mob)
+    xi.darkixion.onMobInitialize(mob)
+end
+
+entity.onMobSpawn = function(mob)
+    xi.darkixion.onMobSpawn(mob)
+end
+
+entity.onAdditionalEffect = function(mob, target, damage)
+    xi.darkixion.onAdditionalEffect(mob, target, damage)
+end
+
+entity.onMobWeaponSkillPrepare = function(mob, target)
+    xi.darkixion.onMobWeaponSkillPrepare(mob, target)
+end
+
+entity.onMobWeaponSkill = function(target, mob, skill)
+    xi.darkixion.onMobWeaponSkill(target, mob, skill)
+end
+
+entity.onMobSkillFinished = function(mob, target, skill)
+    xi.darkixion.onMobSkillFinished(mob, target, skill)
+end
+
+entity.onMobRoamAction = function(mob)
+    xi.darkixion.onMobRoamAction(mob)
+end
+
+entity.onMobRoam = function(mob)
+    xi.darkixion.onMobRoam(mob)
+end
+
+entity.onMobEngaged = function(mob, target)
+    xi.darkixion.onMobEngaged(mob, target)
+end
+
+entity.onMobDisengage = function(mob)
+    xi.darkixion.onMobDisengage(mob)
+end
+
+entity.onMobFight = function(mob, target)
+    xi.darkixion.onMobFight(mob, target)
```

Comment:
You can replace these with a setup helper function in the global:

```
local entity = {}

xi.darkixion.setupEntity(entity)

return entity
```

```
-- All the functions you're going to use in the setup helper should be defined before, so they're available

xi.darkixion.setupEntity = function(enitity)
    entity.onMobEngaged = function(mob, target)
        xi.darkixion.onMobEngaged(mob, target)
    end

    entity.onMobDisengage = function(mob)
        xi.darkixion.onMobDisengage(mob)
    end

    -- etc.
end

Reply:
Fancy. I'll play around with it thanks

Reply:
Well I thought it worked, but seems like the server crashes using this trickery. I tried creating local entity in `setupEntity` like so
```
xi.darkixion.setupEntity = function()
    local entity = {}

    entity.onMobDeath = function(mob, player, optParams)
        xi.darkixion.onMobDeath(mob, player, optParams)
    end

    entity.onMobDespawn = function(mob)
        xi.darkixion.onMobDespawn(mob)
    end

    entity.onMobSpawn = function(mob)
        xi.darkixion.onMobSpawn(mob)
    end

    entity.onMobRoam = function(mob)
        xi.darkixion.onMobRoam(mob)
    end

    entity.onMobEngaged = function(mob, target)
        xi.darkixion.onMobEngaged(mob, target)
    end

    entity.onMobDisengage = function(mob)
        xi.darkixion.onMobDisengage(mob)
    end

    return entity
end
```

and did this in the `Dark_Ixion.lua`

```
-----------------------------------
-- Area: West Sarutabaruta [S]
--   NM: Dark Ixion
-----------------------------------
require('scripts/globals/dark_ixion')
-----------------------------------

return xi.darkixion.setupEntity()
```

> The thread 0x39d0 has exited with code 0 (0x0).
Exception thrown at 0x00007FFC11A9CF19 (KernelBase.dll) in xi_map.exe: 0xE24C4A02.
The thread 0x564 has exited with code 0 (0x0).
The thread 0xda70 has exited with code 0 (0x0).
The thread 0x8864 has exited with code 0 (0x0).
The thread 0x8910 has exited with code 0 (0x0).
The thread 0x17d8 has exited with code 0 (0x0).
The thread 0x5cd4 has exited with code 0 (0x0).
The thread 0xa7dc has exited with code 0 (0x0).
The thread 0xe920 has exited with code 0 (0x0).
The thread 0xa460 has exited with code 0 (0x0).
The thread 0xc33c has exited with code 0 (0x0).
The thread 0x3f8 has exited with code 0 (0x0).
The thread 0x771c has exited with code 0 (0x0).
The thread 0x25f4 has exited with code 0 (0x0).
The thread 0xf3cc has exited with code 0 (0x0).
The thread 0xb598 has exited with code 0 (0x0).
The thread 0x7f04 has exited with code 0 (0x0).
The thread 0xd414 has exited with code 0 (0x0).
The thread 0x8954 has exited with code 0 (0x0).
Exception thrown at 0x00007FFC11A9CF19 (KernelBase.dll) in xi_map.exe: 0xE24C4A02.

Reply:
To be clear I also tried your method first (the code is currently in this PR using that original method)

Reply:
Disregard ... it was an issue with `mob:atPoint()`. Not sure why I couldn't capture with the Visual Studio debugger, but all working now.